draft-ietf-privacypass-auth-scheme.md

```diff
@@ -454,77 +454,92 @@ token, e.g., the probability that one can forge an authenticator value without
-indistinguishable from the issuance protocol failing for arbitrary reasons
-(excluding what can be inferred from latency between the client and origin interaction).
-Likewise, origins SHOULD randomly choose to not challenge clients for tokens
-with some non-zero probability. Moreover, origins SHOULD include random token
-types, from the Reserved list of "greased" types (defined in {{token-types}}),
-with some non-zero probability.
+# Client Behavior {#client-behavior}
+
+When a client receives one or more token challenges in response to a request,
+the client has a set of choices to make:
+
+- Whether or not to redeem a token via a new request to the origin.
+- Whether to use a cached token, or fetch one or more new tokens using
+   the issuance protocol.
+- If multiple challenges were sent, which challenge to use for redeeming a
+  token on a subsequent request.
+
+The approach to these choices depends on the use case of the application, as
+well as the deployment model (see {{Section 4 of ARCHITECTURE}} for discussion
+of the different deployment models).
+
+## Choosing to Redeem Tokens
+
+Some applications of tokens might require clients to always present a token
+as authentication in order to successfully make requests. For example, a restricted
+service that wants to only allow access to valid users, but do so without learning
+specific user Identities, could use tokens that are based on attesting user
+credentials. In these kinds of use cases, clients will need to always redeem a
+token in order to successfully make a request.
+
+Many other use cases for Privacy Pass tokens involve open services that interact
+with some clients that either cannot redeem tokens, or can only sometimes redeem
+tokens. For example, a service can use tokens as a way to reduce the incidence of
+presenting CAPTCHAs to users. In such use cases that are meant for open services,
+clients will regularly encounter challenges for which they cannot redeem a token; and,
+even clients that could redeem a token can choose not to. In order to mitigate the risk
+of these services relying on always receiving tokens, clients can ignore token
+challenges (and thus not redeem a token on a subsequent request) with some
+non-zero probability. See {{Section 5.1 of ARCHITECTURE}} for further considerations
+on avoiding discriminatory behavior across clients when using Privacy Pass tokens.
+
+Clients might also choose to not redeem tokens in subsequent requests when the
+token challenges indicate erroneous or malicious behavior on the part of the
+challenging origin. For example, if a client’s ability to generate tokens via an
+attester and issuer is limited to a certain rate, a malicious origin could send
+an excessive number of token challenges with unique redemption contexts
+in order to cause the client to exhaust its ability to generate new tokens, or
+to overwhelm issuance servers. The limits here will vary based on the specific
```

Comment:
Reading this, I am reminded that it the potential for token hoarding and trading also produces a desire to rate limit issuance.  I went looking for something in the architecture draft and came up short.  The best I could do was [Section 7.1](https://datatracker.ietf.org/doc/html/draft-ietf-privacypass-architecture#section-7.1).  Is that worth spending some text on?

Reply:
Marking as resolved since we now have text on that